### ISSUE TITLE
Form Recognizer: Move to Cognitive Error Model

### ISSUE THREAD

**Issue by annelo-msft** [Azure/azure-sdk-for-net#9702]
4. Common Cognitive/Azure Error Model    
Maybe move ErrorDetails to Azure.Core
Status: No action yet.
Priority: low
Effort: low

This will need to happen in two stages:
- Follow design in TextAnalytics
- Once we land the [error types for Azure.Core](https://github.com/Azure/azure-sdk-for-net/issues/9703), move to this model.

**Comment by annelo-msft** [comment]
@ctstone

**Comment by maririos** [comment]
As we don't have a common error design proposal yet, could we move this issue out of Preview1?

**Comment by annelo-msft** [comment]
Yes, thanks!

**Comment by maririos** [comment]
No longer needed as new clients in FR are using `ResponseError` from Azure.Core
